src/controllers/admin/skill-category-controller.ts

```diff
@@ -12,3 +15,83 @@ export const index = async (req: Request, res: Response) => {
     res.status(500).json({ message: "Something went wrong" })
   }
 }
+/**
+ * Show single category
+ */
+export const show = async (req: Request, res: Response) => {
+  try {
+    const { id } = req.params
+    const skillCategory = await SkillCategoryService.show(parseInt(id))
+    res.json(skillCategory)
+  } catch (error) {
+    if (error instanceof ValidationError) {
+      return res.status(400).json(error)
+    }
+    if (error instanceof CustomError) {
+      return res.status(error.status).json({ message: error.message })
+    }
+    res.status(500).json({ message: "Something went wrong" })
+  }
+}
+
+/**
+ * store new skill category
+ */
+export const store = async (req: Request, res: Response) => {
+  try {
+    const { name, sequence_no, description, status } = req.body
+    await createSkillCategorySchema.validate({ name, sequence_no, description, status })
+    const newSkillCategory = await SkillCategoryService.create({
+      name,
+      sequence_no,
+      description,
+      status,
+    })
+    res.json(newSkillCategory)
+  } catch (error) {
+    res.status(500).json({ message: error })
+  }
+}
+/**
+ * Update skill category
+ */
+export const update = async (req: Request, res: Response) => {
+  try {
+    const { id } = req.params
+    const { sequence_no, ...updatedDataWithoutSequenceNo } = req.body
```

Comment:
extract the params here from req.body, for example `{ name, description } = req.body`

Reply:
done :3